lib/db_utils.go

```diff
@@ -1644,111 +1689,840 @@ func DBGetAllUserGroupEntries(handle *badger.DB, ownerPublicKey []byte) ([]*Mess
+	messagingGroupMember *AccessGroupMember, groupOwnerPublicKey *PublicKey, groupKeyName *GroupKeyName) error {
+
+	if err := DBSetWithTxn(txn, snap, _dbKeyForGroupEnumerationIndex(
+		groupOwnerPublicKey, groupKeyName, messagingGroupMember.GroupMemberPublicKey),
+		EncodeToBytes(blockHeight, nil)); err != nil {
+
+		return errors.Wrapf(err, "DBPutMessagingGroupMemberInMembershipIndexWithTxn: Problem setting messaging recipient with key (%v) "+
+			"and entry (%v) in the db", _dbKeyForGroupEnumerationIndex(
+			groupOwnerPublicKey, groupKeyName, messagingGroupMember.GroupMemberPublicKey),
+			EncodeToBytes(blockHeight, nil))
+	}
+
+	return nil
+}
+
+func DBPutMessagingGroupMemberInEnumerationIndex(handle *badger.DB, snap *Snapshot, blockHeight uint64,
+	messagingGroupMember *AccessGroupMember, groupOwnerPublicKey *PublicKey, groupKeyName *GroupKeyName) error {
+
+	return handle.Update(func(txn *badger.Txn) error {
+		return DBPutMessagingGroupMemberInEnumerationIndexWithTxn(txn, snap, blockHeight, messagingGroupMember, groupOwnerPublicKey, groupKeyName)
+	})
+}
+
+func DBGetAllGroupMembersForMessagingGroup(handle *badger.DB, snap *Snapshot,
+	groupOwnerPublicKey *PublicKey, groupKeyName *GroupKeyName) ([]*AccessGroupMember, error) {
+
+	var messagingGroupMembers []*AccessGroupMember
+
+	err := handle.View(func(txn *badger.Txn) error {
+		var err error
+		messagingGroupMembers, err = DBGetAllGroupMembersForMessagingGroupWithTxn(txn, snap, groupOwnerPublicKey, groupKeyName)
+		return err
+	})
+
+	return messagingGroupMembers, err
+}
+
+// Get all the group members for a given messaging group.
+func DBGetAllGroupMembersForMessagingGroupWithTxn(txn *badger.Txn, snap *Snapshot,
+	groupOwnerPublicKey *PublicKey, groupKeyName *GroupKeyName) ([]*AccessGroupMember, error) {
+	// enumerate all the group members for a given messaging group.
+	var messagingGroupMembers []*AccessGroupMember
+	prefix := _dbSeekPrefixForGroupEnumerationIndex(groupOwnerPublicKey, groupKeyName)
+	keysFound, _, err := _enumerateKeysForPrefixWithTxn(txn, prefix)
+	if err != nil {
+		return nil, errors.Wrapf(err, "DBGetAllGroupMembersForMessagingGroupWithTxn: "+
+			"problem enumerating messaging key entries for prefix (%v)", prefix)
+	}
+
+	for _, keyBytes := range keysFound {
+		// The key is of the form <prefix, GroupOwnerPublicKey, GroupKeyName, GroupMemberPublicKey>
+		// So we extract the GroupEnumerationKey.
+		enumerationKey := &GroupEnumerationKey{}
+		rr := bytes.NewReader(keyBytes)
+		if exists, err := DecodeFromBytes(enumerationKey, rr); !exists || err != nil {
+			return nil, errors.Wrapf(err, "DBGetAllGroupMembersForMessagingGroupWithTxn: "+
+				"problem decoding key (%v)", keyBytes)
+		}
+		memberPublicKey := &enumerationKey.GroupMemberPublicKey
+		messagingGroupMember := DBGetMemberFromMembershipIndexWithTxn(txn, snap, memberPublicKey, groupOwnerPublicKey, groupKeyName)
```

Comment:
could we store `AccessGroupMember` as the value in this index so we don't have to do a second DB lookup?

Reply:
Yes, definitely. Just thought we could save on storage especially when groups or channels might have million+ members! We're also not even calling this function anywhere yet as there is rarely a need to get ALL members. 

Should we still go ahead and store AccessGroupMember as the value here @lazynina @AeonSw4n ?